daktari/test_config.py

```diff
@@ -1,76 +1,153 @@
-import sys
+import os
 import unittest
 from io import StringIO
 from pathlib import Path
+from unittest.mock import patch
 
 from colors import red, yellow
 from packaging import version
 from packaging.version import Version
 
 from daktari import __version__
-from daktari.config import check_version_compatibility, parse_raw_config
 from daktari.version_utils import sanitise_version_string
+from daktari.checks.intellij_idea import IntelliJIdeaInstalled, IntelliJProjectImported
+from daktari.checks.misc import EnvVarSet
+from daktari.config import check_version_compatibility, parse_raw_config, LOCAL_CONFIG_PATH, Config, apply_local_config
+from daktari.resource_utils import get_resource
 
 config_path = Path("./.daktari.config")
 current_version: Version = Version(__version__)
 next_version = version.parse(f"{current_version.major}.{current_version.minor}.{current_version.micro + 1}")
 
+TEST_CHECKS = [
+    EnvVarSet(variable_name="SOME_ENV_VAR"),
+    IntelliJIdeaInstalled(),
+    IntelliJProjectImported(),
+]
 
-class TestConfig(unittest.TestCase):
-    def setUp(self):
-        self.capturedOutput = StringIO()
-        sys.stdout = self.capturedOutput
 
+class TestConfig(unittest.TestCase):
     def tearDown(self):
-        sys.stdout = sys.__stdout__
+        if os.path.exists(LOCAL_CONFIG_PATH):
+            os.remove(LOCAL_CONFIG_PATH)
 
     def test_empty_version(self):
-        result = check_version_compatibility(config_path, "")
-        self.assertEqual(result, True)
-        self.verify_warning(
-            "⚠️  No minimum version found in .daktari.config. Specifying daktari_version is "
-            "recommended to ensure team members have compatible versions installed."
-        )
+        with patch("sys.stdout", new=StringIO()) as fake_out:
```

Comment:
Just wondering why we're doing with patch now, instead of the `setUp` stuff we were doing before. Is this way better?

Reply:
Yeah the `patch` approach is neater I think - the main reason I changed it was that the old approach was giving me test failures depending on how I ran the test suite (PyCharm vs command line). This approach works consistently and also avoids shared test state/globals which I always try to avoid if I can! Bit more boilerplatey, though, admittedly.